resources/views/fact-gedcom-fields.phtml

```diff
@@ -58,16 +59,24 @@ foreach ($keys as $key) {
         <?php $id = Registry::idFactory()->id() ?>
         <?php $expanded = $tree->getPreference('EXPAND_SOURCES') === '1' ?>
         <div>
-            <button type="button" class="btn btn-text p-0" href="#<?= e($id) ?>" data-bs-toggle="collapse" aria-controls="<?= e($id) ?>" aria-expanded="<?= $expanded ? 'true' : 'false' ?>">
-                <?= view('icons/expand') ?>
-                <?= view('icons/collapse') ?>
-             </button>
+			<?php if (Auth::isMember($tree) || ($tree->getPreference('SHOW_SOURCE_DETAIL_TOGGLES') === '0' && $tree->getPreference('SHOW_SOURCE_TITLES') === '0')) : ?>
+				<button type="button" class="btn btn-text p-0" href="#<?= e($id) ?>" data-bs-toggle="collapse" aria-controls="<?= e($id) ?>" aria-expanded="<?= $expanded ? 'true' : 'false' ?>">
+					<?= view('icons/expand') ?>
+					<?= view('icons/collapse') ?>
+				</button>
+			<?php else: ?>
+				<?php $expanded = '' ?>
+			<?php endif ?>
 
             <?php $label = '<span class="label">' . I18N::translate('Source') . '</span>' ?>
-            <?php $value = '<span class="field" dir="auto"><a href="' . e($source->url()) . '">' . $source->fullName() . '</a></span>' ?>
+			<?php if (Auth::isMember($tree) || $tree->getPreference('SHOW_SOURCE_TITLES') === '0') : ?>
+				<?php $value = '<span class="field" dir="auto"><a href="' . e($source->url()) . '">' . $source->fullName() . '</a></span>' ?>
+			<?php else: ?>
+				<?php $value = '<span class="field" dir="auto">' . I18N::translate('PRIVATE') . '</span>' ?>
```

Comment:
if you use `I18N::translate('Private')` instead of `I18N::translate('PRIVATE')` the existing translations to other langguages will be successful ;-)